packages/fuels-accounts/src/predicate.rs

```diff
@@ -40,22 +40,23 @@ impl Predicate {
     }
 
     pub fn set_provider(&mut self, provider: Provider) -> Result<&mut Self> {
-        // Check that the consensus parameters are identical
-        match &self.provider {
-            Some(p) if p.consensus_parameters()? != provider.consensus_parameters()? => {
-                Err(error!(
-                    InvalidData,
-                    "Trying to set a new provider that uses different consensus parameters"
-                ))
-            }
-            _ => {
-                self.provider = Some(provider);
-                Ok(self)
-            }
+        self.address = Self::calculate_address(&self.code, &provider.consensus_parameters());
+        self.provider = Some(provider);
+        Ok(self)
+    }
+
+    pub fn calculate_address(code: &[u8], params: &ConsensusParameters) -> Bech32Address {
+        fuel_tx::Input::predicate_owner(code, params).into()
+    }
+
+    fn consensus_parameters(&self) -> ConsensusParameters {
+        match &self.provider() {
+            None => ConsensusParameters::default(),
+            Some(p) => p.consensus_parameters(),
```

Comment:
A nit, seems more concise:
```suggestion
        self.provider()
            .map(|p| p.consensus_parameters())
            .unwrap_or_default()
```